core/node/api_server/src/web3/namespaces/debug.rs

```diff
@@ -71,6 +104,63 @@ impl DebugNamespace {
         }
     }
 
+    fn map_flatten_call(
+        call: Call,
+        calls: &mut Vec<DebugCallFlat>,
+        trace_address: &mut Vec<usize>,
+        only_top_call: bool,
+        transaction_position: usize,
+        transaction_hash: H256,
+    ) {
+        let subtraces = call.calls.len();
+        let debug_type = match call.r#type {
+            CallType::Call(_) => DebugCallType::Call,
+            CallType::Create => DebugCallType::Create,
+            CallType::NearCall => unreachable!("We have to filter our near calls before"),
+        };
+
+        let result = if call.error.is_none() {
+            Some(CallResult {
+                output: web3::Bytes::from(call.output),
+                gas_used: U256::from(call.gas_used),
+            })
+        } else {
+            None
+        };
+
+        calls.push(DebugCallFlat {
+            action: Action {
+                call_type: debug_type,
+                from: call.from,
+                to: call.to,
+                gas: U256::from(call.gas),
+                value: call.value,
+                input: web3::Bytes::from(call.input),
+            },
+            result,
+            subtraces,
+            traceaddress: trace_address.clone(), // Clone the current trace address
```

Comment:
BTW, is this naming correct? [The PR](https://github.com/ethereum/go-ethereum/pull/26377/files) in go-ethereum seems to use `traceAddress` naming. I.e., the field should be named `trace_address` with `#[serde(rename_all)]` enabled.